File: filelist/src/main/scala/farjs/filelist/popups/HelpController.scala
package farjs.filelist.popups

import farjs.filelist.FileListUiData
import farjs.ui.popup._
import farjs.ui.theme.Theme
import scommons.react._

import scala.scalajs.js

object HelpController extends FunctionComponent[FileListUiData] {

  private[popups] var messageBoxComp: UiComponent[MessageBoxProps] = MessageBox

  protected def render(compProps: Props): ReactElement = {
    val props = compProps.wrapped
    val theme = Theme.useTheme.popup

    if (props.showHelpPopup) {
      <(messageBoxComp())(^.plain := MessageBoxProps(
        title = "Help",
        message = "//TODO: show help/about info",
        actions = js.Array(MessageBoxAction.OK { () =>
          props.onClose()
        }),
        style = theme.regular
      ))()
    }
    else null
  }
}

File: filelist/popups/HelpController.mjs
/**
 * @typedef {import("../FileListUi.mjs").FileListUiData} FileListUiData
 */
import React from "react";
import MessageBox from "@farjs/ui/popup/MessageBox.mjs";
import MessageBoxAction from "@farjs/ui/popup/MessageBoxAction.mjs";
import Theme from "@farjs/ui/theme/Theme.mjs";

const h = React.createElement;

/**
 * @param {FileListUiData} props
 */
const HelpController = (props) => {
  const { messageBoxComp } = HelpController;

  const theme = Theme.useTheme().popup;

  return props.showHelpPopup
    ? h(messageBoxComp, {
        title: "Help",
        message: "//TODO: show help/about info",
        actions: [
          MessageBoxAction.OK(() => {
            props.onClose();
          }),
        ],
        style: theme.regular,
      })
    : null;
};

HelpController.displayName = "HelpController";
HelpController.messageBoxComp = MessageBox;

export default HelpController;
